rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1828,6 +1828,451 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
+                    cashPrice: currentRatePrice -> cashPrice,
+                    datedValue: empty
+                }
+
+	alias changeQuantity:	<"Create distinct list of Quantity with value set to newAssetQuantity">
+			tradeState -> trade -> tradableProduct -> tradeLot only-element -> priceQuantity -> quantity
+				       extract  NonNegativeQuantitySchedule {
+                            value: newAssetQuantity,
+                            unit: unit,
+                                    ...
+                        }
+				then distinct
+
+			
+	alias newPriceQuantity: <"Create the new price quantity object.">
+        PriceQuantity {
+            price: [ newPrice ],
+            quantity: empty,
+            ...
+        }
+
+	set instruction -> split -> breakdown:
+		[ Create_TerminationInstruction( tradeState ) ]
+	
+
+
+        add instruction -> split -> breakdown: <"Sets the second part of the split to be a new contract with a new asset payout.">
+                    [
+                        PrimitiveInstruction {
+                            quantityChange: QuantityChangeInstruction {
+                                    change: [newPriceQuantity],
+                                    direction: QuantityChangeDirectionEnum -> Replace,
+                                    lotIdentifier: empty
+                                },
+                            termsChange: Create_RepriceOrAdjustmentTermsChangeInstruction(
+                                        tradeState -> trade -> tradableProduct -> product -> contractualProduct,
+                                        effectiveRepriceDate
+                                    ),
+                            ...
+                        }
+                    ]
+		
+		
+		
+func Qualify_Reprice: <"This qualification function is used to qualify repricing of a contractual product with an interest rate payout and assetPayout.">
+	[qualification BusinessEvent]
+	inputs:
+		businessEvent BusinessEvent (1..1)
+	output: is_event boolean (1..1)
+	
+	alias beforeTradePrice: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> tradeLot->priceQuantity
+	alias afterTradePrice: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> tradeLot->priceQuantity
+	
+	alias beforeEconomicterms: businessEvent -> instruction only-element -> before -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+	alias afterEconomicTerms: FilterOpenTradeStates( businessEvent -> after ) only-element -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms
+
+	set is_event:
+		businessEvent -> after -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> assetPayout exists and
+		businessEvent -> after -> trade -> tradableProduct -> product -> contractualProduct -> economicTerms -> payout -> interestRatePayout exists and
```

Comment:
I assume you're testing whether the product is a repo, but you're using the representation where `assetPayout` is in `payout` instead of `collateral`, which is inconsistent with the samples also provided. Consider using `Qualify_RepurchaseAgreement` (or `Qualify_SecuritiesFinance` once it supersedes it) to avoid any inconsistency.

Reply:
Removed AssetPayout test